### PR TITLE
feat: add initial bot structure

### PR DESCRIPTION
**Pull request by AgataLa**

### CODE REVIEW

File: PlantHelperBot/PlantHelper/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# files generated during the lubuild process
+generated/
+bin/
+obj/
```
Comment (legeof008) [inline PlantHelperBot/PlantHelper/.gitignore:4]: Just a small thing, but would you mind adding an enplane in here ? Most CI-CD runners are created to require a space here.

File: PlantHelperBot/PlantHelper/Controllers/BotController.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            IBot bot,
+            ILogger<BotController> logger)
+        {
+            _bot = bot ?? throw new ArgumentNullException(nameof(bot));
+            _logger = logger;
+
+            var adapterSettings = configuration.GetSection(AdapterSettings.AdapterSettingsKey).Get<List<AdapterSettings>>() ?? new List<AdapterSettings>();
+            adapterSettings.Add(AdapterSettings.CoreBotAdapterSettings);
+
+            foreach (var adapter in adapters ?? throw new ArgumentNullException(nameof(adapters)))
```
Comment (legeof008) [inline PlantHelperBot/PlantHelper/Controllers/BotController.cs:36]: I'm not particularly familiar with C# or .NET but is the argument thrown here just a runtime exception ? Do we handle it somehow ?

File: PlantHelperBot/PlantHelper/Controllers/BotController.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+        [HttpPost]
+        [HttpGet]
+        [Route("api/{route}")]
+        public async Task PostAsync(string route)
+        {
+            if (string.IsNullOrEmpty(route))
+            {
+                _logger.LogError($"PostAsync: No route provided.");
+                throw new ArgumentNullException(nameof(route));
```
Comment (legeof008) [inline PlantHelperBot/PlantHelper/Controllers/BotController.cs:55]: The same situation as before is here. In SpringBoot there is an annotation that rejects empty query parameters with 404 error in response.  I'm just curious if that's a thing in .NET.

File: PlantHelperBot/PlantHelper/Controllers/BotController.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+                }
+
+                // Delegate the processing of the HTTP POST to the appropriate adapter.
+                // The adapter will invoke the bot.
+                await adapter.ProcessAsync(Request, Response, _bot).ConfigureAwait(false);
+            }
+            else
+            {
+                _logger.LogError($"PostAsync: No adapter registered and enabled for route {route}.");
+                throw new KeyNotFoundException($"No adapter registered and enabled for route {route}.");
```
Comment (legeof008) [inline PlantHelperBot/PlantHelper/Controllers/BotController.cs:72]: Here as well.

File: PlantHelperBot/PlantHelper/Nuget.config [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+      <configuration>
+        <packageSources>
+          <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
+          <add key="BotBuilder.myget.org" value="https://botbuilder.myget.org/F/botbuilder-v4-dotnet-daily/api/v3/index.json" protocolVersion="3" />
+        </packageSources>
+      </configuration>
```
Comment (legeof008) [inline PlantHelperBot/PlantHelper/Nuget.config:7]: Endline please

File: PlantHelperBot/PlantHelper/PlantHelper.csproj [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
+  </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="3.1.8" />
+    <PackageReference Include="Microsoft.Bot.Builder.AI.Luis" Version="4.18.1" />
+    <PackageReference Include="Microsoft.Bot.Builder.AI.QnA" Version="4.18.1" />
+    <PackageReference Include="Microsoft.Bot.Builder.Dialogs.Adaptive.Runtime" Version="4.18.1" />
+  </ItemGroup>
+</Project>
```
Comment (legeof008) [inline PlantHelperBot/PlantHelper/PlantHelper.csproj:19]: Endline please

File: PlantHelperBot/PlantHelper/PlantHelper.dialog [hunk 7]
```diff
@@ -0,0 +1,161 @@
+{
```
Comment (legeof008) [inline PlantHelperBot/PlantHelper/PlantHelper.dialog:1]: Is this file a part of pre-build or pre-run config ?

File: PlantHelperBot/PlantHelper/Program.cs [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+namespace PlantHelper
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            CreateHostBuilder(args).Build().Run();
+        }
+
+        public static IHostBuilder CreateHostBuilder(string[] args) =>
```
Comment (legeof008) [inline PlantHelperBot/PlantHelper/Program.cs:16]: Does this method need to be public ?

File: PlantHelperBot/PlantHelper/Program.cs [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+                    builder.AddBotRuntimeConfiguration(applicationRoot, settingsDirectory, environmentName);
+
+                    builder.AddCommandLine(args);
+                })
+                .ConfigureWebHostDefaults(webBuilder =>
+                {
+                    webBuilder.UseStartup<Startup>();
+                });
+    }
+}
```
Comment (legeof008) [inline PlantHelperBot/PlantHelper/Program.cs:33]: Endline please

File: PlantHelperBot/PlantHelper/PlantHelper.botproj [hunk 10]
```diff
@@ -0,0 +1,5 @@
+{
+  "$schema": "https://raw.githubusercontent.com/microsoft/BotFramework-Composer/main/Composer/packages/server/schemas/botproject.schema",
+  "name": "PlantHelper",
+  "skills": {}
+}
```
Comment (legeof008) [inline PlantHelperBot/PlantHelper/PlantHelper.botproj:5]: Endline please

File: PlantHelperBot/PlantHelper/Properties/launchSettings.json [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+    "BotProject": {
+      "commandName": "Project",
+      "launchBrowser": true,
+      "applicationUrl": "http://localhost:3978",
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
+      }
+    }
+  }
+}
```
Comment (legeof008) [inline PlantHelperBot/PlantHelper/Properties/launchSettings.json:27]: Endline please

File: PlantHelperBot/PlantHelper/Startup.cs [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+namespace PlantHelper
+{
+    public class Startup
+    {
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
```
Comment (legeof008) [inline PlantHelperBot/PlantHelper/Startup.cs:19]: Does this mean that the variable is private within the scope of the package but has a public getter ?

**Review by legeof008 — CHANGES_REQUESTED** [review]
Outside of code I have questions regarding the whole structure as a whole:

- Are these .lu files needed ?
- Are the png and pdf files needed ?

**Review by legeof008 — APPROVED** [review]

**Review by legeof008 — APPROVED** [review]

**Review by patrykferenc — APPROVED** [review]

**Pull request merged**